torch/CMakeLists.txt

```diff
@@ -356,6 +356,9 @@ if(USE_UCC)
   target_compile_definitions(torch_python PRIVATE USE_UCC)
 endif()
 
+# Allow torch_python to link to OpenTelemetry API headers
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../third_party/opentelemetry-cpp/api/include)
```

Comment:
This is wrong, please use `target_include_directories` to make it available only to a specific targets (say `libtorch_cpu`)

Also, don't you need a minimum runtime that comes with it? Or is it a header only file?

Reply:
@malfet I don't see why a Find module is needed, if we are vendoring. If opentelemetry doesn't occur in any public headers we don't need to distribute it either.

Reply:
I've made changes to the code to follow some of the suggested concepts. As far as I understood @malfet - the main concern raised was that if someone/somewhere wants to use local OTEL headers instead of ones coming from `third_party`. I did add a `FindOpenTelemetry.cmake` in the latest commit - but can remove it - if it doesn't make sense?
LMK @malfet / @ezyang